### PR TITLE
Fix user management create/edit flows

### PR DESCRIPTION
**Pull request by vishal423**
Replace ngModel occurrence with corresponding reactive form control, update tests

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by avdev4j** [comment]
Hi @vishal423 
Could you please provide some details about the goal of this PR ?

thanks

**Comment by vishal423** [comment]
@avdev4j, done.

**Comment by wmarques** [comment]
For me we shouldn't use the reactive form as the field isn't editable... That's why there was a readonly ngModel

**Comment by vishal423** [comment]
IMO, It's bad design to mix `ngModel` and reactive forms in the same component. Irrespective of design choice, this code needs to be corrected to fix following console error:

> ERROR 
> Error: 
>       ngModel cannot be used to register form controls with a parent formGroup directive.  Try using
>       formGroup's partner directive "formControlName" instead.  Example:
>     <div [formGroup]="myGroup">
>       <input formControlName="firstName">
>     </div>
>     In your class:
>     this.myGroup = new FormGroup({
>        firstName: new FormControl()
>     });
>       Or, if you'd like to avoid registering this form control, indicate that it's standalone in ngModelOptions:
>       Example:
>     <div [formGroup]="myGroup">
>        <input formControlName="firstName">
>        <input [(ngModel)]="showMoreControls" [ngModelOptions]="{standalone: true}">
>     </div>
>

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management-update.component.ts.ejs [hunk 1]
```diff
@@ -98,6 +100,7 @@ export class UserMgmtUpdateComponent implements OnInit {
     }
 
     private updateUser(user: User): void {
+        user.id = this.editForm.get(['id']).value;
```
Comment (wmarques) [inline generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management-update.component.ts.ejs]: is it necessary as we won't update the user ID ?

**Comment by wmarques** [comment]
@vishal423 oh yes you are correct about the error, let's do this this way then

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management-update.component.ts.ejs (hunk 1 above)
Comment (vishal423) [reply generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management-update.component.ts.ejs]: not really. Will update later today

**Pull request merged**